lib/buffer.js

```diff
@@ -20,6 +20,8 @@ function createPool() {
   poolOffset = 0;
 }
 
+// v8::String::kMaxLength defined in v8.h
+const kStringMaxLength = (1 << 28) - 16;
```

Comment:
Can I suggest putting this in `lib/internal/buffer.js` instead of exposing it to user land?

@skomski You can access it in the by adding `// Flags: --expose-internals` at the top, then `require('internal/buffer')`.